devtools/maven/src/main/java/io/quarkus/maven/QuarkusBootstrapProvider.java

```diff
@@ -205,14 +225,15 @@ protected CuratedApplication doBootstrap(QuarkusBootstrapMojo mojo, LaunchMode m
 
             final ApplicationModel appModel;
             try {
-                appModel = modelResolver.resolveManagedModel(artifactCoords, Collections.emptyList(), managingProject(mojo),
+                appModel = modelResolver.resolveManagedModel(artifactCoords, forcedExtensions, managingProject(mojo),
                         reloadableModules);
             } catch (AppModelResolverException e) {
                 throw new MojoExecutionException("Failed to bootstrap application in " + mode + " mode", e);
             }
 
             QuarkusBootstrap.Builder builder = QuarkusBootstrap.builder()
                     .setAppArtifact(appModel.getAppArtifact())
+                    .setForcedDependencies(forcedExtensions)
```

Comment:
Oh, the current impl is actually broken. The thing is `forcedDependencies(...)` is actually called a few lines overriding what you are setting but it doesn't actually matter because the model is resolved a few lines above.